ash/src/extensions/khr/video_queue.rs

```diff
@@ -0,0 +1,182 @@
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_video_session_khr)(
+            self.handle,
+            video_session,
+            allocation_callbacks.as_raw_ptr(),
+        );
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkDestroyVideoSessionParametersKHR.html>
+    pub unsafe fn destroy_video_session_parameters(
+        &self,
+        video_session_parameters: vk::VideoSessionParametersKHR,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_video_session_parameters_khr)(
+            self.handle,
+            video_session_parameters,
+            allocation_callbacks.as_raw_ptr(),
+        );
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetVideoSessionMemoryRequirementsKHR.html>
+    pub unsafe fn get_video_session_memory_requirements(
+        &self,
+        video_session: vk::VideoSessionKHR,
+    ) -> VkResult<Vec<vk::VideoSessionMemoryRequirementsKHR<'_>>> {
+        read_into_defaulted_vector(|count, data| {
+            (self.fp.get_video_session_memory_requirements_khr)(
+                self.handle,
+                video_session,
+                count,
+                data,
+            )
+        })
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkUpdateVideoSessionParametersKHR.html>
+    pub unsafe fn update_video_session_parameters(
+        &self,
+        video_session_parameters: vk::VideoSessionParametersKHR,
+        update_info: &vk::VideoSessionParametersUpdateInfoKHR<'_>,
+    ) -> VkResult<()> {
+        (self.fp.update_video_session_parameters_khr)(
+            self.handle,
+            video_session_parameters,
+            update_info,
+        )
+        .result()
+    }
+}
+
+impl crate::khr::video_queue::Instance {
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetPhysicalDeviceVideoCapabilitiesKHR.html>
+    pub unsafe fn get_physical_device_video_capabilities(
+        &self,
+        physical_device: vk::PhysicalDevice,
+        video_profile: &vk::VideoProfileInfoKHR<'_>,
+        capabilities: &mut vk::VideoCapabilitiesKHR<'_>,
+    ) -> VkResult<()> {
+        (self.fp.get_physical_device_video_capabilities_khr)(
+            physical_device,
+            video_profile,
+            capabilities,
+        )
+        .result()
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetPhysicalDeviceVideoFormatPropertiesKHR.html>
+    pub unsafe fn get_physical_device_video_format_properties(
+        &self,
+        physical_device: vk::PhysicalDevice,
+        video_format_info: &vk::PhysicalDeviceVideoFormatInfoKHR<'_>,
+    ) -> VkResult<Vec<vk::VideoFormatPropertiesKHR<'_>>> {
+        read_into_defaulted_vector(|count, data| {
```

Comment:
Since the user could technically request more structures to be filled by use of the `pNext` chain in `VkVideoFormatPropertiesKHR`, this should be split between a `_len()` function and the actual getter that takes `&mut [vk::VideoFormatPropertiesKHR<'_>]`. There are various examples of this around the code-base to copy.